contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -244,8 +244,8 @@ pub fn execute_vote(
         return Err(ContractError::InvalidVote {});
     }
 
-    if prop.current_status(&env.block)? != Status::Open {
-        return Err(ContractError::NotOpen { id: proposal_id });
+    if prop.expiration.is_expired(&env.block) {
```

Comment:
ah I also realized that if the proposal is not expired, but it has either been `Closed` or `Executed`, we probably should not allow voting. I don't think there's anything technically wrong with that, but it seems unintuitive. We should probably also explicitly check for the states `Open`, `Passed`, or `Rejected` here.